MSAL/src/instance/MSALAuthority.m

```diff
@@ -157,7 +157,9 @@ + (void)resolveEndpointsForAuthority:(NSURL *)unvalidatedAuthority
         tenant = firstPathComponent;
     }
     
-    MSALAuthority *authorityInCache = [MSALAuthority authorityFromCache:updatedAuthority userPrincipalName:userPrincipalName];
+    MSALAuthority *authorityInCache = [MSALAuthority authorityFromCache:updatedAuthority
+                                                          authorityType:authorityType
+                                                      userPrincipalName:userPrincipalName];
     
```

Comment:
Should we also respect validate argument here? We're currently doing validation even when validate == NO. Otherwise this argument is unnecessary.